pkg/htmx/gin.go

```diff
@@ -55,12 +58,37 @@ func GetHxCurrentPath(ctx *gin.Context) string {
 }
 
 func HasMatchingParentPath(ctx *gin.Context) bool {
-	currentUrlParts := strings.Split(GetHxCurrentPath(ctx), "/")
-	shortCurrent := path.Join(currentUrlParts[0 : len(currentUrlParts)-1]...)
-	requestUrlParts := strings.Split(ctx.Request.URL.Path, "/")
-	shortRequest := path.Join(requestUrlParts[0 : len(requestUrlParts)-1]...)
-	fmt.Println(shortCurrent, shortRequest)
-	return shortCurrent == shortRequest
+	currentPath := GetHxCurrentPath(ctx)
+	currentPathParts := strings.Split(currentPath, "/")
+
+	requestPath := ctx.Request.URL.Path
+	requestPathParts := strings.Split(requestPath, "/")
+	requestPathParent := path.Join(requestPathParts[0 : len(requestPathParts)-1]...)
+
+	// If the currentPath is a parent, then check that against the requestedPath's parent
+	if len(currentPathParts) == len(requestPathParts)-1 {
+		return strings.TrimPrefix(currentPath, "/") == requestPathParent
+	}
+
+	// Else, check the equality of parent paths
+	currentPathParent := path.Join(currentPathParts[0 : len(currentPathParts)-1]...)
+	return currentPathParent == requestPathParent
+}
+
+func IsRequestingParentPath(ctx *gin.Context) bool {
+	currentPath := GetHxCurrentPath(ctx)
+	currentPathParts := strings.Split(currentPath, "/")
+	currentPathParent := path.Join(currentPathParts[0 : len(currentPathParts)-1]...)
+
+	requestPath := ctx.Request.URL.Path
+	requestPathParts := strings.Split(requestPath, "/")
+
+	// If the requestPath is a parent, then check that against the currentPath's parent
+	return len(currentPathParts)-1 == len(requestPathParts) && currentPathParent == strings.TrimPrefix(requestPath, "/")
+}
```

Comment:
A previous commit had this logic inside the above `HasMatchingParentPath` func, but I discovered that it broke rendering a new sidenav when navigating via sidenav anchor tags because those two paths would have the same number of parts and thus check against their parent which will always be the same. 

Breaking this out into a function that only checks for the requesting path to be a parent resolves the above and keeps the functionaliy of navigating to a parent and skipping the render of the sidenav component.